build_runner_core/test/generate/build_test.dart

```diff
@@ -1171,5 +1171,29 @@ void main() {
           outputs: {},
           writer: writer);
     });
+
+    test('the entrypoint cannot be read by a builder', () async {
+      var builders = [
+        applyToRoot(new TestBuilder(
+            buildExtensions: replaceExtension('.txt', '.hasEntrypoint'),
+            build: (buildStep, _) async {
+              var hasEntrypoint = await buildStep
+                  .findAssets(new Glob('**'))
+                  .contains(
+                      makeAssetId('a|.dart_tool/build/entrypoint/build.dart'));
+              await buildStep.writeAsString(
```

Comment:
presumably you could just literally put an `expect` in here? but this is fine too :)

Reply:
yeah, would have to wrap the entire thing with an `expectAsync2` as well.